src/main/java/io/vertx/ext/sql/SQLRowStream.java

```diff
@@ -0,0 +1,33 @@
+package io.vertx.ext.sql;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.streams.ReadStream;
+
+/**
+ * A ReadStream of Rows from the underlying RDBMS. This class follows the ReadStream semantics and will automatically
+ * close the underlying resources if all returned rows are returned. For cases where the results are ignored before the
+ * full processing of the returned rows is complete the close method **MUST** be called in order to release underlying
+ * resources.
+ *
+ * The interface is minimal in order to support all SQL clients not just JDBC.
+ */
+@VertxGen
+public interface SQLRowStream extends ReadStream<JsonArray> {
+
+    /**
+     * Will convert the column name to the json array index
+     *
+     * @param name
+     * @return
+     */
+    int column(String name);
```

Comment:
Columns names are even case insensitive (by default) so they will cause trouble once you convert to json object too, this method solves that issue.

What do you mean by cursor size? that you want to receive a list of rows? because that will make the usage more complex since you receive a json array and then you need to see if there are more arrays inside then you're handling more than 1 row per call (cursor size) otherwise it is just a row... and then it gets even more tricky if your query returns a single column of type array which then you don't know if it's supposed to be a cursor sized list or a single row...